cheroot/test/test_ssl.py

```diff
@@ -775,31 +811,61 @@ def test_http_over_https_error(
     if ip_addr is ANY_INTERFACE_IPV6:
         fqdn = '[{fqdn}]'.format(**locals())
 
-    expect_fallback_response_over_plain_http = adapter_type == 'pyopenssl'
-    if expect_fallback_response_over_plain_http:
-        resp = requests.get(
-            f'http://{fqdn!s}:{port!s}/',
-            timeout=http_request_timeout,
-        )
-        assert resp.status_code == 400
-        assert resp.text == (
-            'The client sent a plain HTTP request, '
-            'but this server only speaks HTTPS on this port.'
-        )
-        return
+    resp = requests.get(
+        f'http://{fqdn!s}:{port!s}/',
+        timeout=http_request_timeout,
+    )
+    assert resp.status_code == 400
+    assert resp.text == (
+        'The client sent a plain HTTP request, '
+        'but this server only speaks HTTPS on this port.'
+    )
 
-    with pytest.raises(requests.exceptions.ConnectionError) as ssl_err:
-        requests.get(  # FIXME: make stdlib ssl behave like PyOpenSSL
-            f'http://{fqdn!s}:{port!s}/',
-            timeout=http_request_timeout,
-        )
 
-    underlying_error = ssl_err.value.args[0].args[-1]
-    err_text = str(underlying_error)
-    assert underlying_error.errno == expected_error_code, (
-        'The underlying error is {underlying_error!r}'.format(**locals())
-    )
-    assert expected_error_text in err_text
+@pytest.mark.parametrize(
+    ('error', 'should_raise', 'error_msg'),
+    (
+        (OSError(32, 'Broken pipe'), False, None),  # EPIPE - ignorable
+        (
+            OSError(9999, 'An error to reckon with'),  # not ignorable
+            True,
+            'An error to reckon with',
+        ),
+    ),
+    ids=['ignorable_error', 'non_ignorable_error'],
```

Comment:
there's a kebab-case + tuple convention for IDs
```suggestion
    ids=('ignorable-error', 'non-ignorable-error'),
```